### ISSUE TITLE
Joined / chained queries with missing lock statuses not detected

### ISSUE THREAD

**Issue by tnoppa** [Riverside-Software/sonar-openedge#992]
As title says joined / chained queries with missing lock statuses are not detected with the current prolint ruleset for sonarqube.
Example.
```
FOR EACH Customer WHERE Customer.State = "Alabama" NO-LOCK,
        FIRST Order WHERE Order.Customer = Customer.CustomerID:

END.

```
 This did not trigger the share-lock rule in sonarqube, but the old way and version of prolint does find this and notify about the share-lock.
Sonarqube finds the share-lock, if the joined/chained search is split into separate ones.

**Comment by gquerret** [comment]
Confirmed.

**Comment by gquerret** [comment]
Fixed in develop
